eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -78,45 +78,45 @@ jobs:
-                SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
-                EnvVars:
-                  Pool: $(Pool)
-                  ${{ insert }}: ${{ parameters.EnvVars }}
-        - ${{ if not(parameters.TestResourceDirectories) }}:
-          - template: /eng/common/TestResources/deploy-test-resources.yml
+          - template: /eng/common/TestResources/build-test-resource-config.yml
             parameters:
-              ${{ if or(parameters.Location, parameters.CloudConfig.Location) }}:
-                Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
-              ServiceDirectory: '${{ parameters.ServiceDirectory }}'
-              SubscriptionConfiguration: $(SubscriptionConfiguration)
-              ArmTemplateParameters: $(ArmTemplateParameters)
-              UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
-              ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
-              SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+              SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
+              SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
               EnvVars:
                 Pool: $(Pool)
                 ${{ insert }}: ${{ parameters.EnvVars }}
+              ${{ if parameters.UseFederatedAuth }}:
+                SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+
+          - ${{ if parameters.TestResourceDirectories }}:
+              - ${{ each directory in parameters.TestResourceDirectories }}:
+                  - template: /eng/common/TestResources/deploy-test-resources.yml
+                    parameters:
+                      ? ${{ if or(parameters.Location, parameters.CloudConfig.Location) }}
+                      : Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
+                      ServiceDirectory: "${{ directory }}"
+                      SubscriptionConfiguration: $(SubscriptionConfiguration)
+                      ArmTemplateParameters: $(ArmTemplateParameters)
+                      UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
+                      ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
+                      SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+                      EnvVars:
+                        Pool: $(Pool)
+                        ${{ insert }}: ${{ parameters.EnvVars }}
+          - ${{ if not(parameters.TestResourceDirectories) }}:
+              - template: /eng/common/TestResources/deploy-test-resources.yml
+                parameters:
+                  ? ${{ if or(parameters.Location, parameters.CloudConfig.Location) }}
```

Comment:
What is this `?` syntax I've never seen this in devops yaml before.

Reply:
This is caused by formatting which appears to be a configuration in the JS repo interacting with some set of plugins (and their configurations) on my machine. I'll revert the formatting.